File: xpenbox-back/src/main/java/org/xpenbox/account/dto/AccountDeactivateRequestDTO.java
package org.xpenbox.account.dto;

import io.quarkus.runtime.annotations.RegisterForReflection;
import jakarta.validation.constraints.Size;

/**
 * Data Transfer Object for deactivating an Account.
 * @param accountResourceCodeToTransfer Resource code of the account to transfer remaining balance to.
 */
@RegisterForReflection
public record AccountDeactivateRequestDTO(

    @Size(min = 1, max = 100, message = "Account resource code for transfer to must be between 1 and 100 characters")
    String accountResourceCodeToTransfer
) { }

File: xpenbox-mobile/src/app/feature/account/model/account.deactivate.request.dto.ts

/**
 * Data Transfer Object for account deactivation request.
 * Includes the resource code of the account to which remaining balance should be transferred.
 */
export class AccountDeactivateRequestDTO {
    accountResourceCodeToTransfer?: string;

    constructor(accountResourceCodeToTransfer: string | undefined) {
        this.accountResourceCodeToTransfer = accountResourceCodeToTransfer;
    }
}